src/autowiring/CoreContext.cpp

```diff
@@ -300,81 +301,78 @@ void CoreContext::AddInternal(const AnySharedPointer& ptr) {
 
   // Verify that this type isn't already satisfied
   MemoEntry& entry = m_typeMemos[ptr.type()];
-  if (entry.m_value)
+  if (entry.m_local && entry.m_value)
     throw autowiring_error("This interface is already present in the context");
 
   // Now we can satisfy it:
+  entry.m_local = true;
   entry.m_value = ptr;
   UpdateDeferredElement(std::move(lk), entry);
 }
 
-MemoEntry& CoreContext::FindByType(AnySharedPointer& reference, bool localOnly) const {
+MemoEntry& CoreContext::FindByType(auto_id type, bool nonrecursive) const {
```

Comment:
This breaks compatibility with an existing use case.

    std::shared_ptr<WinUSBPayloadReader> rdr;
    streaming->FindByType<WinUSBPayloadReader>(rdr);

Given we can't use auto_id there, what is the replacement? Also, would this push the version to 0.8.0?